app/src/main/java/com/prangesoftwaresolutions/audioanchor/PlayActivity.java

```diff
@@ -321,12 +336,18 @@ private void playAudio() {
         startService();
         Intent broadcastIntent = new Intent(Broadcast_PLAY_NEW_AUDIO);
         sendBroadcast(broadcastIntent);
+
+        if(mSleepTimer != null)
+            mSleepTimer.startTimer(false);
     }
 
     private void pauseAudio() {
         //Send a broadcast to the service -> PLAY_NEW_AUDIO
         Intent broadcastIntent = new Intent(Broadcast_PAUSE_AUDIO);
         sendBroadcast(broadcastIntent);
+
+        if(mSleepTimer != null)
+            mSleepTimer.restartTimer();
```

Comment:
I personally find it annoying if I just want to quickly pause the audio but then I have to restart the sleep timer. If I pause the audio and don't continue to listen, there is no problem if the sleep timer continues. For other cases, I have the option to cancel the sleep timer by pressing the back button in the PlayActivity or entering 0 in the sleep timer alert dialog.
Therefore I would not restart the sleep timer when pausing, but either A) leave it running as it was previously, or B) pause it as well and continue the count down when the playback is continued.
I personally prefer option A) but we can discuss this if you have a different opinion.

Reply:
Let's just keep it at A) for now - I'll revert it for this commit. I think the feature is kind of half-baked anyway, because pause is the only action that resets the timer.
Maybe we can eventually add a option that would make all actions (including moving forward/back, seeking etc.), reset the timer. That's how I personally prefer it anyway. But that would be for another PR :).